packages/siri-sx-generator/index.ts

```diff
@@ -1,16 +1,45 @@
-import { GetObjectCommand, S3Client } from "@aws-sdk/client-s3";
+import { GetObjectCommand, S3Client, PutObjectCommand } from "@aws-sdk/client-s3";
 import { S3Event } from "aws-lambda";
+import { toXML } from "jstoxml";
 
 const s3Client = new S3Client({ region: "eu-west-2" });
 
+const uploadToS3 = async (data: string, filename: string) => {
+    // Unique bucket name
+    const bucketName = process.env.SIRI_SX_BUCKET_NAME || "";
+    // Name for uploaded object
+    const keyName = filename;
+
+    const putCommand: PutObjectCommand = new PutObjectCommand({
+        Bucket: bucketName,
+        Key: keyName,
+        Body: data,
+    });
+
+    await s3Client.send(putCommand);
+    console.log("Successfully uploaded data to " + bucketName + "/" + keyName);
+};
+
 export const main = async (event: S3Event): Promise<void> => {
     const bucketName = event.Records[0].s3.bucket.name || "";
     const objectKey = event.Records[0].s3.object.key || "";
 
     const params = { Bucket: bucketName, Key: objectKey };
     const response = await s3Client.send(new GetObjectCommand(params));
-    const data = response.Body?.toString() || "";
+    const data = (await response.Body?.transformToString()) || "";
+    const config = {
+        indent: "    ",
+    };
+
+    if (!data) {
+        throw Error("No data found");
+    }
+
+    const xmlData = toXML(JSON.parse(data), config);
+
+    if (!xmlData) {
```

Comment:
Does this ever happen? I would have thought that the 'toXML' function throws an error if it cant convert the data to XML, instead of setting it to undefined, but I dont know how the library works.

Reply:
couldn't find it in docs so this is precautionary

Reply:
export function toXML(obj?: XmlElement | XmlElement[], options?: XmlOptions): string;

Reply:
It seems like it would either return a string or error, but it cant hurt having this check in. No worries.